src/native/eventpipe/ds-protocol.c

```diff
@@ -453,6 +453,43 @@ ds_ipc_message_try_parse_uint32_t (
 	return result;
 }
 
+bool
+ds_ipc_message_try_parse_string_utf16_t_string_utf8_t_alloc (
+	uint8_t **buffer,
+	uint32_t *buffer_len,
+	const ep_char8_t **string_utf8)
```

Comment:
having an alloc function returning const sounds like incorrect ownership pattern, how will you free something you can't write?

Reply:
I thought originally that we would just internally document that the EventPipeProviderConfiguration would own the data, and just use `const` to ensure that callers cannot modify/free unless explicitly recasting. But after this thread https://github.com/dotnet/runtime/pull/115265#discussion_r2149671922 changed it to not const